packages/core/core/src/services/document-service/collection-type.ts

```diff
@@ -15,119 +18,159 @@ export const createCollectionTypeRepository: RepositoryFactoryMethod<Schema.Coll
 
   async function findMany(params = {} as any) {
     // TODO: replace with chaining
-    DP.defaultToDraft(params);
-    DP.statusToLookup(params);
-    i18n.defaultLocale(contentType, params);
-    i18n.localeToLookup(contentType, params);
-
-    return documents.findMany(uid, params);
+    const queryParams = await pipeAsync(
+      DP.defaultToDraft,
+      DP.statusToLookup,
+      i18n.defaultLocale(contentType),
+      i18n.localeToLookup(contentType)
+    )(params);
+
+    return documents.findMany(uid, queryParams);
   }
 
   async function findFirst(params = {} as any) {
-    DP.defaultToDraft(params);
-    DP.statusToLookup(params);
-    i18n.defaultLocale(contentType, params);
-    i18n.localeToLookup(contentType, params);
-
-    return documents.findFirst(uid, params);
+    const queryParams = await pipeAsync(
+      DP.defaultToDraft,
+      DP.statusToLookup,
+      i18n.defaultLocale(contentType),
+      i18n.localeToLookup(contentType)
+    )(params);
+
+    return documents.findFirst(uid, queryParams);
   }
 
   async function findOne(id: string, params = {} as any) {
-    DP.defaultToDraft(params);
-    DP.statusToLookup(params);
-    i18n.defaultLocale(contentType, params);
-    i18n.localeToLookup(contentType, params);
-
-    return documents.findOne(uid, id, params);
+    const queryParams = await pipeAsync(
+      DP.defaultToDraft,
+      DP.statusToLookup,
+      i18n.defaultLocale(contentType),
+      i18n.localeToLookup(contentType)
+    )(params);
+
+    return documents.findOne(uid, id, queryParams);
   }
 
   async function deleteFn(id: string, params = {} as any) {
-    DP.statusToLookup(params);
-    i18n.localeToLookup(contentType, params);
-
-    return documents.delete(uid, id, params);
-  }
+    const queryParams = await pipeAsync(omit('status'), i18n.localeToLookup(contentType))(params);
 
-  async function deleteMany(params = {} as any) {
-    return documents.deleteMany(uid, params);
+    return documents.delete(uid, id, queryParams);
   }
 
   async function create(params = {} as any) {
-    DP.setStatusToDraft(params);
-    DP.statusToData(params);
-    DP.filterDataPublishedAt(params);
-    i18n.defaultLocale(contentType, params);
-    i18n.localeToData(contentType, params);
+    const queryParams = await pipeAsync(
+      DP.setStatusToDraft,
+      DP.statusToData,
+      DP.filterDataPublishedAt,
+      i18n.defaultLocale(contentType),
+      i18n.localeToData(contentType)
+    )(params);
+
+    const doc = await documents.create(uid, queryParams);
+
+    if (params.status === 'published') {
```

Comment:
Could you explain this a bit? I can't see how status would end up as 'published'
based on how we're piping the params above.

Also if it is 'published' would it be better to use documents.publish()?

Reply:
Sure 

- We create a draft by default as you see in the queryParams (setStatusToDraft)
- The idea is to allow if the user had send status = 'published' to auto publish
- We do a new create because the publish action is actually trying to delete & does a lot of work we know we don't need during create. -> I plan to improve the document-engine code more in the upcoming iterations create more consistency